docs/source/en/model_doc/gemma2.md

```diff
@@ -16,34 +16,138 @@ rendered properly in your Markdown viewer.
 
+<Tip>
+Click the right sidebar's Gemma 2 models for additional task examples.
+</Tip>
+
+The example below demonstrates how to generate text based on an image with [`Pipeline`] or the [`AutoModel`] class.
+
+<hfoptions id="usage">
+<hfoption id="Pipeline">
+
+
+### Text Generation with `Pipeline`
+
+```python
+from transformers import pipeline
+import torch
+
+pipe = pipeline(
+    "text-generation",
+    model="google/gemma-2-9b-it",
+    model_kwargs={"torch_dtype": torch.bfloat16},
+    device="cuda",
+)
+
+messages = [
+    {"role": "user", "content": "Explain quantum computing simply"},
+]
+outputs = pipe(messages, max_new_tokens=256)
+print(outputs[0]["generated_text"])
+```
+### Text Generation with `AutoModel`
+```python
+from transformers import AutoTokenizer, AutoModelForCausalLM
+import torch
+
+tokenizer = AutoTokenizer.from_pretrained("google/gemma-2-9b-it")
+model = AutoModelForCausalLM.from_pretrained(
+    "google/gemma-2-9b-it",
+    device_map="auto",
+)
+
+input_text = "Write me a poem about Machine Learning."
+input_ids = tokenizer(input_text, return_tensors="pt").to("cuda")
+
+outputs = model.generate(**input_ids, max_new_tokens=32)
+print(tokenizer.decode(outputs[0]))
+```
+### Using `transformers-cli`
+```
+echo -e "Plants create energy through a process known as" | transformers-cli run --task text-generation --model google/gemma-2-2b --device 0 
+```
+
+### Quantized version through `bitsandbytes`
+
+Quantization reduces model size and speeds up inference by converting high-precision numbers (e.g., 32-bit floats) to lower-precision formats (e.g., 8-bit integers), with minimal accuracy loss
+#### Using 8-bit precision (int8)
+
+```python
+from transformers import AutoTokenizer, AutoModelForCausalLM, BitsAndBytesConfig
+
+quantization_config = BitsAndBytesConfig(load_in_8bit=True)
+
+tokenizer = AutoTokenizer.from_pretrained("google/gemma-2-27b-it")
+model = AutoModelForCausalLM.from_pretrained(
+    "google/gemma-2-27b-it",
+    quantization_config=quantization_config,
+)
+
+input_text = "Write me a poem about Machine Learning."
+input_ids = tokenizer(input_text, return_tensors="pt").to("cuda")
+
+outputs = model.generate(**input_ids, max_new_tokens=32)
+print(tokenizer.decode(outputs[0]))
+```
+#### Using 4-bit precision
+```python
+from transformers import AutoTokenizer, AutoModelForCausalLM, BitsAndBytesConfig
+
+quantization_config = BitsAndBytesConfig(load_in_4bit=True)
+
+tokenizer = AutoTokenizer.from_pretrained("google/gemma-2-27b-it")
+model = AutoModelForCausalLM.from_pretrained(
+    "google/gemma-2-27b-it",
+    quantization_config=quantization_config,
+)
+
+input_text = "Write me a poem about Machine Learning."
+input_ids = tokenizer(input_text, return_tensors="pt").to("cuda")
+
+outputs = model.generate(**input_ids, max_new_tokens=32)
+print(tokenizer.decode(outputs[0]))
+
+```
+### AttentionMaskVisualizer
+
+```python
+visualizer = AttentionMaskVisualizer("google/gemma-2b")
```

Comment:
```suggestion
from transformers.utils.attention_visualizer import AttentionMaskVisualizer

visualizer = AttentionMaskVisualizer("google/gemma-2b")
```

Reply:
Can you print out the image, upload it to https://huggingface.co/datasets/huggingface/documentation-images/tree/main/transformers/model_doc (ping me to merge!), and then add it here?